app/controllers/graders_controller.rb

```diff
@@ -135,9 +135,12 @@ def global_actions
       positions = params[:criteria]
       # TODO: simplify data format interface between here and Criterion#assign_tas.
       criterion_ids_types =
-        @assignment.rubric_criteria.where(position: positions).pluck(:id).map { |id| [id, 'RubricCriterion'] } +
-          @assignment.flexible_criteria.where(position: positions).pluck(:id).map { |id| [id, 'FlexibleCriterion'] } +
-          @assignment.checkbox_criteria.where(position: positions).pluck(:id).map { |id| [id, 'CheckboxCriterion'] }
+        @assignment.criteria.where(type: 'RubricCriterion')
```

Comment:
@david-yz-liu i assume we can get rid of this line and replace it with criterion ids (and refactor criterion `assign_tas` to only handle a parameter of criterion ids)?

Reply:
Yes exactly, where before we needed "criterion id + criterion type" we now only need "criterion id"